packaging/build-container.sh

```diff
@@ -0,0 +1,70 @@
+#! /bin/bash -e
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+EXEC_PATH="$(dirname "$(realpath "$0")")"
+PROJECT_PATH="$(dirname $EXEC_PATH)"
+
+DEFAULT_BUILD_FLAGS="--no-cache --rm"
+DEFAULT_TAG_PREFIX="nmstate"
+
+options=$(getopt --options "" \
+    --longoptions extra-args: \
+    -- "${@}")
+
+
+eval set -- "${options}"
+while :
+do
+    case "${1}" in
+        --extra-args)
+            shift
+            extra_args="${1}"
+            ;;
+        --)
+            shift
+            break
+            ;;
+    esac
+    shift
+done
+
+rebuild_container() {
+    local container_name
+    local extra_args
+
+    extra_args="${1}"
+    shift
+
+    # remove leading tag prefix
+    container_name="${1#*/}"
+
+    # remove container name suffix
+    echo "${1}" | grep -q "/" && : tag_prefix="${1%/*}"
```

Comment:
I do not understand what this does exactly.

Reply:
If $1 contains a slash, it will store everything before the slash in `tag_prefix`.